aws/data_source_aws_mks_cluster.go

```diff
@@ -0,0 +1,60 @@
+package aws
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsMksCluster() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsMksClusterRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"broker_node_group_info": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"encrypt_rest_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enhanced_monitoring": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"kafka_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"broker_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsMksClusterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kafkaconn
+	arn := d.Get("arn").(string)
```

Comment:
This is currently optional so seems like a poor choice to be used to read the MSK cluster state from AWS. Either it needs to be mandatory or you need to rework this data source and the `readMskClusterState` to look up the cluster by name (which is going to be much more useful than asking people to pass an ARN into a data source).

If this isn't easily possible, I'd probably ditch the data source for now and it can be added in a later pull request.